gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -183,6 +192,17 @@ def __init__(
     self._split_alternate_allele_info_fields = (
         split_alternate_allele_info_fields)
     self._annotation_field_set = set(annotation_fields or [])
+    if counter_factory:
```

Comment:
Considering that we do want to keep the NoOp counters, what do you think of creating a NoOpCounterFactory as well that returns the NoOpCounter? You can then just have a line `counter_factory = counter_factory or NoOpCounterFactory()` and you don't need two cases anymore. This pattern can be applied to any other module that wants to use counters as well. P.S. "by induction", you can make NoOpCounter private as well in this pattern :)

Reply:
I have already done this in the third PR because dealing with null objects was getting too ugly there. But did not want to add it back here since you asked for its removal :-) Now that we are on the same page even here, I added it back.